File: node/transcribe_file_async_separate_recognition.js
const { SpeechClient } = require("@soniox/soniox-node");
const { asyncSleep } = require("./utils");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  console.log("Uploading file.");
  const file_id = await speechClient.transcribeFileAsync(
    "../test_data/test_audio_multi_channel.flac",
    "test", // reference_name
    {
      num_audio_channels: 2,
      enable_separate_recognition_per_channel: true,
    }
  );
  console.log(`File ID: ${file_id}`);

  let status;
  let error_message;

  while (true) {
    console.log("Calling getTranscribeAsyncStatus.");
    const response = await speechClient.getTranscribeAsyncStatus(file_id);
    status = response.status;
    error_message = response.error_message;

    console.log(`Status: ${status}`);

    if (status === "COMPLETED" || status === "FAILED") {
      break;
    }

    await asyncSleep(2000);
  }

  if (status === "COMPLETED") {
    const channel_results = await speechClient.getTranscribeAsyncResult(
      file_id
    );
    for (result of channel_results) {
      console.log(
        `Channel ${result.channel}: ${result.words
          .map((word) => word.text)
          .join(" ")}`
      );
    }
  } else {
    console.log(`Transcription failed with error: ${error_message}`);
  }

  console.log("Calling deleteTranscribeAsyncFile.");
  await speechClient.deleteTranscribeAsyncFile(file_id);
})();

File: speech_to_text_legacy/python/transcribe_file_async_separate_recognition.py
import time
from soniox.speech_service import SpeechClient
from soniox.transcribe_file import transcribe_file_async


def main():
    with SpeechClient() as client:
        print("Uploading file.")
        file_id = transcribe_file_async(
            "../test_data/test_audio_multi_channel.flac",
            client,
            model="en_v2",
            num_audio_channels=2,
            enable_separate_recognition_per_channel=True,
            reference_name="test",
        )
        print(f"File ID: {file_id}")

        while True:
            print("Calling GetTranscribeAsyncStatus.")
            status = client.GetTranscribeAsyncStatus(file_id)
            print(f"Status: {status.status}")
            if status.status in ("COMPLETED", "FAILED"):
                break
            time.sleep(2.0)

        if status.status == "COMPLETED":
            print("Calling GetTranscribeAsyncResult")
            channel_results = client.GetTranscribeAsyncResult(file_id)
            for result in channel_results:
                print(f"Channel {result.channel}: " + "".join(word.text for word in result.words))
        else:
            print(f"Transcription failed with error: {status.error_message}")

        print("Calling DeleteTranscribeAsyncFile.")
        client.DeleteTranscribeAsyncFile(file_id)


if __name__ == "__main__":
    main()
